src/io.jl

```diff
@@ -182,10 +181,9 @@ function _load(png_ptr, info_ptr; gamma::Union{Nothing,Float64}=nothing, expand_
         buffer_eltype = UInt8
     end
     
-    is_gray = (color_type == PNG_COLOR_TYPE_GRAY_ALPHA) | (color_type == PNG_COLOR_TYPE_GRAY)
-    if _check_background_load(background, is_gray, valid_PLTE, read_as_paletted)
-        is_transparent || throw(ArgumentError("Only transparent images can have a background"))
+    if read_with_background
         background_color = process_background(png_ptr, info_ptr, _adjust_background_bitdepth(background, bit_depth))
+        is_transparent || @warn("Background color for non-transparent image: $(background_color)")
```

Comment:
Does the user need to be warned about this? If not, perhaps make it a debug?

Reply:
I just saw your message above. Yeah, I don't think this needs to be generate log noise. Being a debug would at least make it accessible until we expose an api for it